src/panels/lovelace/cards/hui-gauge-card.js

```diff
@@ -0,0 +1,199 @@
+          transform-origin: center top;
+          transition: all 1.3s ease-in-out;
+        }
+        .gauge-data{
+          z-index: 4;
+          color: var(--primary-text-color);
+          line-height: calc(var(--base-unit) * 0.3);
+          position: absolute;
+          width: calc(var(--base-unit) * 4);
+          height: calc(var(--base-unit) * 2.1);
+          top: calc(var(--base-unit) * 1.2);
+          margin-left: auto;
+          margin-right: auto;
+          transition: all 1s ease-out;
+        }
+        .gauge-data #percent{
+          font-size: calc(var(--base-unit) * 0.55);
+        }
+        .gauge-data #title{
+          padding-top: calc(var(--base-unit) * 0.15);
+          font-size: calc(var(--base-unit) * 0.30);
+        }
+        .not-found {
+          flex: 1;
+          background-color: yellow;
+          padding: 8px;
+        }
+      </style>
+      <ha-card  on-click='_handleClick'>
+        <div class='container'>
+          <div class='gauge-a'></div>
+          <div class='gauge-b'></div>
+          <div class='gauge-c' id='gauge'></div>
+          <div class='gauge-data'>
+            <div id='percent'>[[_computeStateDisplay(_stateObj)]]</div>
+            <div id='title'>[[_computeTitle(_stateObj)]]</div>
+          </div>
+        </div>
+      </ha-card>
+    `;
+  }
+
+  static get properties() {
+    return {
+      hass: {
+        type: Object
+      },
+      _config: Object,
+      _stateObj: {
+        type: Object,
+        computed: '_computeStateObj(hass.states, _config.entity)',
+        observer: '_stateObjChanged'
+      },
+    };
+  }
+
+  getCardSize() {
+    return 1;
+  }
+
+  setConfig(config) {
+    if (!config || !config.entity) throw new Error('Invalid card configuration');
+    this._config = Object.assign({ min: 0, max: 100 }, config);
+  }
+
+  _computeStateObj(states, entityId) {
+    return states && entityId in states ? states[entityId] : null;
+  }
+
+  _stateObjChanged(stateObj) {
+    if (!stateObj || isNaN(stateObj.state)) return;
+
+    const config = this._config;
+    const turn = this._translateTurn(stateObj.state, config) / 10;
+
+    this.$.gauge.style.transform = `rotate(${turn}turn)`;
+    this.$.gauge.style.backgroundColor = this._computeSeverity(stateObj.state, config.severity);
+  }
+
+  _computeStateDisplay(stateObj) {
+    if (!stateObj || isNaN(stateObj.state)) return '';
+    const unitOfMeasurement = this._config.unit_of_measurement || stateObj.attributes.unit_of_measurement || '';
+    return `${stateObj.state} ${unitOfMeasurement}`;
+  }
+
+  _computeTitle(stateObj) {
+    if (!stateObj) {
+      this.$.title.className = 'not-found';
+      return 'Entity not available: ' + this._config.entity;
+    }
+    if (isNaN(stateObj.state)) {
+      this.$.title.className = 'not-found';
+      return 'Entity is non-numeric: ' + this._config.entity;
+    }
+    return this._config.title;
```

Comment:
We should reset the className

Reply:
btw it's kinda a hack to have a _computeFunction have side effects, should actually be done in _stateChanged, however it's good enough for initial merge.